src/server/actions/Agencies.ts

```diff
@@ -10,23 +10,33 @@ import {
   ServiceModel,
 } from '@/server/models';
 
+type CurrentStatusFilters = {
+  showCompleted: boolean;
+  showNeedsReview: boolean;
+  showExpired: boolean;
+};
+
 /**
  * @brief Gets all agencies
  * @param populateServices Populates the agencies' "services" field
  * @param searchString Filters agencies that do not have the search term present in their name or
+ * @param currentStatusFilters Filters agencies based on their current status
  * @param compareFn Sorts the array of agencies before returning using the specified function; leave empty for no sorting
  * @returns An array of all agencies in the "agencies" collection
  */
 export async function getAgencies(
   populateServices: boolean = true,
   searchString?: string,
+  currentStatusFilters?: CurrentStatusFilters,
   compareFn?: (a: Agency, b: Agency) => number
 ): Promise<Agency[]> {
   try {
     await dbConnect();
-    let query = searchString
-      ? AgencyModel.find({ name: new RegExp(searchString, 'i') })
-      : AgencyModel.find({});
+    let query = AgencyModel.find({});
+
+    if (searchString) {
+      query = query.find({ name: new RegExp(searchString, 'i') });
+    }
```

Comment:
is there a reason that we need to perform 2 separate queries if there is a search string as opposed to the prior implementation?